src/lib/models/workflow-execution.ts

```diff
@@ -39,41 +21,54 @@ export interface WorkflowExecution {
   taskQueue?: string;
   historyEvents: Long;
   pendingActivities: PendingActivityInfo[];
-}
+  url: string;
+  toggleUrl: (isFullScreen: boolean) => string;
+};
 
 type WorkflowExecutionAPIResponse = Optional<
   DescribeWorkflowExecutionResponse,
   'executionConfig' | 'pendingActivities' | 'pendingChildren'
 >;
 
-export class WorkflowExecution {
-  constructor({
-    executionConfig,
-    workflowExecutionInfo,
-    pendingActivities,
-  }: WorkflowExecutionAPIResponse) {
-    this.name = workflowExecutionInfo.type.name;
-    this.id = workflowExecutionInfo.execution.workflowId;
-    this.runId = workflowExecutionInfo.execution.runId;
-    this.startTime = formatDate(workflowExecutionInfo.startTime);
-    this.endTime = formatDate(workflowExecutionInfo.closeTime);
-    this.status = workflowExecutionInfo.status;
-    this.historyEvents = workflowExecutionInfo.historyLength;
-
-    if (executionConfig) {
-      this.taskQueue = executionConfig.taskQueue.name;
-    }
-
-    this.pendingActivities = pendingActivities;
-  }
-
-  get url(): string {
-    return `/workflows/${this.id}/${this.runId}`;
-  }
+export const toWorkflowExecution = (
+  response: WorkflowExecutionAPIResponse,
+): WorkflowExecution => {
+  const name = response.workflowExecutionInfo.type.name;
+  const id = response.workflowExecutionInfo.execution.workflowId;
+  const runId = response.workflowExecutionInfo.execution.runId;
+  const startTime = formatDate(response.workflowExecutionInfo.startTime);
+  const endTime = formatDate(response.workflowExecutionInfo.closeTime);
+  const status = response.workflowExecutionInfo.status;
+  const historyEvents = response.workflowExecutionInfo.historyLength;
+  const url = `/workflows/${id}/${runId}`;
+  const taskQueue = get(response, 'executionConfig.taskQueue.name');
```

Comment:
as discussed directly, the typescript that is bundled here doesn't yet support the `?.` notation. We may want to get back to such cases as we have newer typescript and replace the `get`s